thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java

```diff
@@ -236,10 +236,17 @@ private void createYamlOnboardingTask(long configId, long tuningWindowStart, lon
     info.setTuningWindowStart(tuningWindowStart);
     info.setTuningWindowEnd(tuningWindowEnd);
     info.setEnd(System.currentTimeMillis());
-    info.setStart(info.getEnd() - ONBOARDING_REPLAY_LOOKBACK);
+
+    long lastTimestamp = detectionConfig.getLastTimestamp();
+    // If no value is present, set the default lookback
+    if (lastTimestamp < 0) {
```

Comment:
Since the default value of lastTimestamp will be 0 (when not set), we should probably change this condition here to not trigger detection on the entire dataset.

Reply:
Hey @akshayrai 
During `org.apache.pinot.thirdeye.detection.yaml.translator.DetectionConfigTranslator#generateDetectionConfig`, the default is always set to `-1` in which case, it is set to `System.currentTimeMillis()`. ref: `org/apache/pinot/thirdeye/detection/yaml/YamlResource.java:242`